pkg/tsdb/schema/schema.go

```diff
@@ -175,15 +176,22 @@ func rateToHours(samplesIngestionRate string) (int, error) {
 	}
 }
 
-func validateAggregatesGranularity(aggregationGranularity string) error {
+func validateAggregatesGranularity(aggregationGranularity string, partitionInterval string, hasAggregates bool) error {
 	dayMillis := 24 * int64(time.Hour/time.Millisecond)
 	duration, err := utils.Str2duration(aggregationGranularity)
 	if err != nil {
 		return err
 	}
 
 	if dayMillis%duration != 0 && duration%dayMillis != 0 {
-		return errors.New("The aggregation granularity should be a divisor or a dividend of 1 day. Examples: \"10m\"; \"30m\"; \"2h\".")
+		return errors.New("the aggregation granularity should be a divisor or a dividend of 1 day. Examples: \"10m\"; \"30m\"; \"2h\".")
+	}
+
+	if hasAggregates {
+		partitionIntervalDuration, _ := utils.Str2duration(partitionInterval) // safe to ignore error since we create 'partitionInterval'
```

Comment:
If it's so safe, I dare you to panic on it 😛

Reply:
i tried real hard, couldn't...

Reply:
Try harder!